toolchain/check/handle_function.cpp

```diff
@@ -333,6 +333,32 @@ static auto RequestVtableIfVirtual(
     CARBON_DIAGNOSTIC(ImplWithoutBase, Error, "impl without base class");
     context.emitter().Emit(node_id, ImplWithoutBase);
   }
+
+  if (generic_id != SemIR::GenericId::None) {
+    bool function_is_generic = true;
```

Comment:
You might try:

`bool function_is_generic =  context.generics().GetSelfSpecific(generic_id) != context.generics().GetSelfSpecific(class_info.generic_id);`

(as the full approach)

Reply:
Per discussion with Richard -- don't do this, it's not intended to work. So either the current approach, or you can compare the bindings_id (without further fetching) from the self_specific_id. But the number of loads is basically the same either way.